ledger/ledgercore/statedelta.go

```diff
@@ -230,3 +468,320 @@ func (sd *StateDelta) OptimizeAllocatedMemory(proto config.ConsensusParams) {
 		sd.Creatables = creatableDeltas
 	}
 }
+
+// GetBasicsAccountData returns basics account data for some specific address
+func (ad NewAccountDeltas) GetBasicsAccountData(addr basics.Address) (basics.AccountData, bool) {
+	idx, ok := ad.acctsCache[addr]
+	if !ok {
+		return basics.AccountData{}, false
+	}
+
+	result := basics.AccountData{}
+	acct := ad.accts[idx].AccountData
+	AssignAccountData(&result, acct)
+
+	if len(ad.appParams) > 0 {
+		result.AppParams = make(map[basics.AppIndex]basics.AppParams)
+		for aapp, val := range ad.appParams {
+			if aapp.Address == addr {
+				if val == nil {
+					delete(result.AppParams, aapp.App)
```

Comment:
if you're building up result.AppParams from empty here, won't you not need to delete the app ID?

Reply:
Good catch, this is leftover from the initial prototype implementation. This is used only in tests now, I'll consider removing this func completely.

Reply:
done